yarn-project/circuits.js/src/structs/private_call_stack_item.ts

```diff
@@ -96,16 +97,16 @@ export class PrivateCallStackItem {
    * Creates a new CallRequest with values of the calling contract.
    * @returns A CallRequest instance with the contract address, caller context, and the hash of the call stack item.
    */
-  public toCallRequest() {
+  public toCallRequest(parentCallContext: CallContext) {
     if (this.isEmpty()) {
       return CallRequest.empty();
     }
 
-    const callContext = this.publicInputs.callContext;
-    const callerContext = callContext.isDelegateCall
-      ? new CallerContext(callContext.msgSender, callContext.storageContractAddress)
+    const currentCallContext = this.publicInputs.callContext;
+    const callerContext = currentCallContext.isDelegateCall
+      ? new CallerContext(parentCallContext.msgSender, parentCallContext.storageContractAddress)
```

Comment:
Why do we need to explicitly pass the `parentCallContext` to it? If it's a delegate call, parent call's `msgSender` and `storageContractAddress` should match those of the call stack item.

Reply:
Because the `callerContractAddress` in callRequest has to be the *actual_for_realsies* (technical term) contract address that called the function, or the kernel checks fail (since for delegateCalls msgSender and storageContractAddress is proxied through the callstack, we cannot use it anymore once we cross the entrypoint). 

And since we needed it anyways, I used it more in this function because the naming is  clearer and we don't have to think about the context propagation anymore: If I need the caller, I look into my parent's context.